lib/src/store.dart

```diff
@@ -199,23 +211,33 @@ class Store<State> {
   // This will be called after all other middleware provided by the user have
   // been run. Its job is simple: Run the current state through the reducer,
   // save the result, and notify any subscribers.
-  _reduceAndNotify(dynamic action) {
+  void _reduceAndNotify(dynamic action) {
     final state = reducer(_state, action);
+
+    if (distinct) {
```

Comment:
Instead of using the `Stream.distinct`, I decided we could do it manually. Why? For broadcast streams (the `onChange` stream is a broadcast Stream), `distinct` will run the `==` check on *every* subscriber. That could become slow / use a lot of memory for large State trees with a lot of subscribers, especially for Mobile / Flutter apps. 

Since we have the two states right here, we can avoid that expensive computation / memory usage by doing this ourselves right in the `_reduceAndNotify` method.

Reply:
> distinct will run the == check for every listener

I'm not familiar with a situation where `==` is slow so I suppose users can use `stream.distinct` at their own risk.

I'm also seeing that `flutter_redux` has to work around distinct for other reasons:

```

    // Don't use `Stream.distinct` because it cannot capture the initial
    // ViewModel produced by the `converter`.
    if (widget.distinct) {
      stream = stream.where((vm) {
        final isDistinct = vm != latestValue;

        return isDistinct;
      });
    }
```

Does this also solve that issue?

Reply:
"I'm not familiar with a situation where == is slow so I suppose users can use stream.distinct at their own risk."

It's a fair point. The reason I thought of it: say you had a couple hundred or even a couple thousand models in your Store. That's really not too crazy -- at SoundCloud, we often held that many models in our Store (peeps have large playlists, scroll through infinite lists, etc).

Seemed like an easy win in that case, but I feel like you might be a bit more suspect about adding this to the library in general? Happy to chat about that if you've got that gut feeling.

-------

WRT `flutter_redux` and the way distinct works for the `StoreConnector` Widget, that's handling a bit of a different case. `StoreConnector.distinct` works on the `ViewModel` returned by the `converter` function, not on the whole State tree. `flutter_redux` also uses a `StreamBuilder` under the hood to handle the subscription stuff. `StreamBuilder` works by first delivering a `Snapshot<ViewModel>` without any data, then delivers a Snapshot with data after it receives data from the Stream.

So what should be rendered before the `onChange` Stream emits something? Why not just build the `ViewModel` based on the `store.state`. However, since you can't pass that first synchronous computation to the `distinct`, I use that small workaround you found :)

Reply:
> It's a fair point. The reason I thought of it: say you had a couple hundred or even a couple thousand models in your Store. That's really not too crazy -- at SoundCloud, we often held that many models in our Store (peeps have large playlists, scroll through infinite lists, etc).

Why would models in your store impact the speed of `==`? Are you referring to checking every instance of every value in the store? I can see that, but wouldn't a basic `==` work to resolve this if your reducers don't mutate state? Ultimately `return state;` means the same object is returned so `==` should be quick, no?

Reply:
Yep, was thinking `identical` would handle most of the cases and be fast. If it did item-by-item comparison I was worried that might be slow, but perhaps it's an unfounded concern.